### ISSUE TITLE
Azure Blob Storage paths are not URL-escaped

### ISSUE THREAD

**Issue by dgolub** [Azure/azure-sdk-for-go#19475]
### Bug Report

Import path: `github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container`
SDK version: `v0.5.1`
Go version: `go1.18.5 windows/amd64`

When opening a blob on Azure Blob Storage with a path that contains a special character (e.g. `foo%5Cbar`), the path doesn't get properly URL-escaped.  It comes back with a not found error, even though a blob with the specified name actually exists.

For example, this fails:
```
blobClient := containerClient.NewBlobClient("foo%5Cbar")
```

But this succeeds:
```
path := url.PathEscape("foo%5Cbar")
blobClient := containerClient.NewBlobClient(path)
```

**Comment by siminsavani-msft** [comment]
Hi @dgolub ! Thank you for reporting an issue! I just wanted to clarify that in your scenario, the blobname is 'foo%5Cbar', is this correct?

**Comment by dgolub** [comment]
Yes, that's correct.

**Comment by siminsavani-msft** [comment]
@dgolub Thank you for your prompt reply! We are working on this and will update this thread when we have additional updates.

**Comment by siminsavani-msft** [comment]
Hi @dgolub ! I am not able to reproduce. Could you please share the failure/error you are getting?

**Comment by ncw** [comment]
I'll just note that the rclone test suite picks up this bug also @siminsavani-msft - its a bug which has been present in Onedrive in the past too! Though I think it is an SDK bug this time since the original Azure blob SDK `github.com/Azure/azure-storage-blob-go/azblob` does **not** have this problem.

**Comment by jhendrixMSFT** [comment]
The fix will roll out in `v0.6.0`.
